### PR TITLE
[Rest Api Compatibility] Typed and xpack graph explore api

### PR DESCRIPTION
**Pull request by pgomulka**
adds back the typed and xpack endpoints for  graph explore api
prevoiusly removed in #46935

relates main meta issue #51816
relates types removal issue #54160

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: x-pack/plugin/graph/src/test/java/org/elasticsearch/xpack/graph/rest/action/RestGraphActionTests.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Before
+    public void setUpAction() {
+        controller().registerHandler(new RestGraphAction());
+        verifyingClient.setExecuteVerifier((actionType, request) -> {
+            assertThat(request, instanceOf(GraphExploreRequest.class));
+            return Mockito.mock(GraphExploreResponse.class);
+        });
+    }
+
+    public void testTypeInPath() {
```
Comment (romseygeek) [inline x-pack/plugin/graph/src/test/java/org/elasticsearch/xpack/graph/rest/action/RestGraphActionTests.java:39]: Maybe add a test for the `_xpath` variant as well?

**Review by romseygeek — APPROVED** [review]
LGTM - I left one comment about testing

File: x-pack/plugin/graph/src/test/java/org/elasticsearch/xpack/graph/rest/action/RestGraphActionTests.java (hunk 1 above)
Comment (pgomulka) [reply x-pack/plugin/graph/src/test/java/org/elasticsearch/xpack/graph/rest/action/RestGraphActionTests.java:39]: good point, in fact no all endpoints are correctly deprecated. /{index}/{type}/_xpack/graph/_explore api was indicating to use /{index}/{type}/_graph/explore
I will update this on 7.x and then update this PR too

**Pull request merged**